### PR TITLE
refactor: Port to 1.18.2

### LINKED ISSUE #197: Bug: Attempting to use Aether biomes in a Superflat world defaults to minecraft:plains.
**(No longer relevant)**
> **Steps to reproduce:**
> 1. Start creating a new superflat world
> 2. Edit the preset, change the default preset biome from `minecraft:plains` to `aether:aether_skylands`
> 3. Click "Done"
> 
> Log: [debug.log](https://github.com/Gilded-Games/The-Aether/files/7198332/debug.log)

### LINKED ISSUE #270: Bug: Dimensions have a static seed at the moment.
This is a Minecraft issue that may need to be fixed either through a Mixin or just not using the json system (which we might be doing in 1.18 anyways). Either that or maybe https://github.com/MinecraftForge/MinecraftForge/pull/7955 will finally be pulled by the time we have to worry about this.

### LINKED ISSUE #312: Enhancement: Change name of default Aether biome.

### LINKED ISSUE #331: Enhancement: Make tree density based on using biomes.
Three different "biomes" will be created with different tree densities, similar to the handling in the original mod.

### LINKED ISSUE #464: Chore: Update mod, dependencies, and mappings.
Should probably do this once Forge 1.18.2 is available.
- Update Forge to latest.
- Update Curios to latest.

### LINKED ISSUE #473: Bug: Custom dimensions using the Aether's dimension effect crash the game
### What Feature Types Apply to This Bug?

World-Gen

### Other Type

_No response_

### What Type of Bug Is This?

Crash

### Forge Version

39.0.88

### The Aether 1 Version

1.18.1-v1.0.0

### Is This Bug a Conflict With Another Mod?

_No response_

### Client Log

_No response_

### Crash Report (if applicable)

_No response_

### Steps to Reproduce

1. Create a new world with the following data pack: [aecrash.zip](https://github.com/Gilded-Games/The-Aether/files/8182222/aecrash.zip)
2. Enter spectator mode
3. /execute in aecrash:i_work_fine run tp ~ ~ ~
4. Note that the game doesn't crash
5. /execute in aecrash:death_test run tp ~ ~ ~
6. Watch as the game hangs for about three minutes, cuts to a screen made of blue dirt and then closes itself

### What You Expect To Happen

An empty dimension with the Aether's dimension effects would be accessed.

### What Actually Happened

We get a crash.

### Additional Details

Crash report not attached since it's probably not necessary - the issue was discussed as being plausible enough in Discord

### Please Read and Confirm The Following

- [X] I have confirmed this bug can be replicated without the use of Optifine.
- [X] I have confirmed this bug is on the most recently supported version of Minecraft.
- [X] I have confirmed the details provided in this report are concise as possible and does not contained vague information (ie. Versions are properly recorded, answers to questions are clear).
- [X] I have confirmed this issue is unique and has not been reported already.

### LINKED ISSUE #475: Feature Request: ultracold dimension_type option, or possibly a dimension tag?
### What Feature Types Apply to This Request?

World-Gen

### Other Type

_No response_

### Is This an Enhancement to a Preexisting Feature?

- [X] I confirm this request is an enhancement to a feature already in The Aether.

### What Is Your Feature Request?

The vanilla dimension_type specification allows for the "ultrawarm" setting to be specified. If set to true, this prevents water from being placed in the dimension, causes wet sponges to be placed as dry sponges, increases lava's flow rate and distance, and changes stalactite dripping behaviour.

The Aether dimension has a similar attribute, in that it forbids lava from being placed in much the same way as the Nether forbids water from being placed. In this case, placed lava immediately becomes aerogel.

The problem with this implementation is that due to it being hardcoded, any custom data pack dimensions cannot make use of this feature, and it remains Aether-exclusive.

To make this available for other dimensions to use in data packs, either of the following could apply:
- dimension_type files would also accept an "ultracold" boolean. This would be false by default (and could also be completely left out of the file, which would set it to false for compatibility's sake). If set to on, lava would immediately become aerogel.
- If this is infeasible, it may be possible to achieve this via tags instead - a tag could be created for dimensions such that any dimensions in it will have all placed lava freeze into aerogel.

### Why Should It Be in the Mod?

Compatibility's sake and functional consistency. For example, a modpack which contains the Aether may also include a new custom dimension with an associated data pack which could make use of this freezing functionality.

### Does This Feature Improve Compatibility With Another Mod?

_No response_

### Please Read and Confirm the Following

- [X] I have confirmed this feature is either in the original Aether mod, or improves parity between The Aether and Minecraft or other mods.
- [X] I have confirmed this suggestion is unique and has not been suggested already.

### LINKED ISSUE #485: Chore: Investigate reusing enqueueWork in 1.18.2.
If it breaks any registries from loading them too late, those registries can be moved to being loaded by the forge registry event to make sure they're loaded earlier on.

### LINKED ISSUE #516: Chore: Can .classpath-old be removed?
It's a 0 byte file that I can't find much info about: https://github.com/Gilded-Games/The-Aether/blob/1.18/.classpath-old.

### PR DESCRIPTION
**Pull request by bconlon1**
- [x] Move Crystal Trees to Surface Structure phase and see if it fixes skyroot trees spawning on their islands.
- [x] Merge 1.18 into this branch again. 
- [x] New tag system is used.
- [x] Comment out structure code for now as we are not registering structures yet.
- [x] Fix DataGen for world-related jsons: Dimension, Dimension Type, Biomes, Noise Settings, Configured Features, Placed Features
- [x] Game freezes when exiting an Aether portal back to the Overworld.
- [x] Closes #464.
- [x] Closes #197.
- [x] Closes #312.
- [x] Closes #331.
- [x] Closes #475.
- [x] Closes #473.
- [x] Closes #270.
- [x] Closes #485.
- [x] Closes #516.

### CODE REVIEW

**Comment by bconlon1** [comment]
There is a new parchment build `v2022.03.06` available, should check if this one fixes the mapping names for stuff.

File: src/main/java/com/gildedgames/aether/common/registry/AetherTags.java [hunk 1]
```diff
@@ -1,124 +1,124 @@
...
-		public static final Tags.IOptionalNamedTag<EntityType<?>> SWET_TARGETS = tag("swet_targets");
-		public static final Tags.IOptionalNamedTag<EntityType<?>> AECHOR_PLANT_TARGETS = tag("aechor_plant_targets");
-
-		private static Tags.IOptionalNamedTag<EntityType<?>> tag(String name) {
-			return EntityTypeTags.createOptional(new ResourceLocation(Aether.MODID, name));
-		}
+		public static final TagKey<EntityType<?>> PIGS = tag("pigs");
+		public static final TagKey<EntityType<?>> NO_SKYROOT_DOUBLE_DROPS = tag("no_skyroot_double_drops");
+		public static final TagKey<EntityType<?>> DEFLECTABLE_PROJECTILES = tag("deflectable_projectiles");
+		public static final TagKey<EntityType<?>> SWET_TARGETS = tag("swet_target");
```
Comment (bconlon1) [inline src/main/java/com/gildedgames/aether/common/registry/AetherTags.java]: Needs to be renamed back to swet_targets.

**Review by Drullkus — APPROVED** [review]
Lookin' good to me!

**Review by Burning-Cactus — APPROVED** [review]
Looks good to me!

**Pull request merged**